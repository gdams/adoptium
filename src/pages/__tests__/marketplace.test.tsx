import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest'
import { axe } from 'vitest-axe';
import Marketplace, { Head } from '../marketplace';
import AxiosInstance from 'axios'
import MockAdapter from 'axios-mock-adapter';

const mock = new MockAdapter(AxiosInstance);

vi.mock('../../util/shuffle', () => {
  return {
    shuffle: (array) => {
      array = [
        {
          name: "Adoptium",
          icon: "vendor-adoptium.png",
          postDownload: null
        }
      ]
      return array
    }
  };
});

afterEach(() => {
  vi.clearAllMocks();
});

describe('Marketplace page', () => {
  it('renders correctly', () => {
    mock.onGet().reply(200, []);

    const { container } = render(<Marketplace />);
    // eslint-disable-next-line
    const pageContent = container.querySelector('main');
    expect(pageContent).toMatchSnapshot();
  });

  it('head renders correctly', () => {
    mock.onGet().reply(200, []);

    const { container } = render(<Head />);
    // eslint-disable-next-line
    const title = container.querySelector('title');
    expect(title?.textContent).toEqual('Marketplace | Adoptium');
  });

  it('has no accessibility violations', async () => {
    mock.onGet().reply(200, []);

    const { container } = render(<Marketplace />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
