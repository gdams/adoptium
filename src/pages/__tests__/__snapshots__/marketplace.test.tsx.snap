// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Marketplace page > renders correctly 1`] = `
<main>
  <section
    class="py-5 text-center container"
  >
    <div
      class="row py-lg-5"
    >
      <div
        class="col-lg-8 col-md-8 mx-auto"
      >
        <h1
          class="fw-light"
        >
          AdoptiumÂ® Marketplace
        </h1>
        <p
          class="pt-3"
        >
          Text
        </p>
        <div
          class="row align-items-center pt-3"
        >
          <div
            class="col-6 col-md-4"
          >
            <img
              alt="AQAvit logo"
              class="img-fluid"
              src="/images/aqavit-light.png"
              width="150"
            />
          </div>
          <div
            class="col-12 col-sm-6 col-md-8 "
          >
            <p
              class="text-start"
            >
              Text
            </p>
          </div>
        </div>
      </div>
    </div>
    <ul
      class="vendor-list pt-5"
    >
      <li
        class="vendor-li"
        data-testid="li-adoptium"
      >
        <input
          checked=""
          class="vendor-name"
          data-testid="checkbox-adoptium"
          id="vendor-adoptium"
          readonly=""
          type="checkbox"
        />
        <label
          class="vendor-label"
          for="vendor-adoptium"
          title="Adoptium"
        >
          <img
            alt="Adoptium icon"
            src="/images/vendors/vendor-adoptium.png"
          />
        </label>
      </li>
    </ul>
    <div
      class="input-group mb-5 row g-2"
    >
      <div
        class="input-group-prepend flex-colunm col-12 col-md-3"
      >
        <label
          class="px-2 fw-bold"
          for="os"
        >
          Text
        </label>
        <select
          aria-label="OS Filter"
          class="form-select form-select-sm"
          data-testid="os-filter"
          id="os-filter"
        >
          <option
            value="any"
          >
            Text
          </option>
          <option
            value="mock_linux"
          >
            Mock_linux
          </option>
          <option
            value="mock_macos"
          >
            Mock_macos
          </option>
          <option
            value="mock_windows"
          >
            Mock_windows
          </option>
        </select>
      </div>
      <div
        class="input-group-prepend flex-colunm col-12 col-md-3"
      >
        <label
          class="px-2 fw-bold"
          for="arch"
        >
          Text
        </label>
        <select
          aria-label="Architecture Filter"
          class="form-select form-select-sm"
          data-testid="arch-filter"
          id="arch-filter"
        >
          <option
            value="any"
          >
            Text
          </option>
          <option
            value="mock_aarch64"
          >
            mock_aarch64
          </option>
          <option
            value="mock_ppc64"
          >
            mock_ppc64
          </option>
          <option
            value="mock_x64"
          >
            mock_x64
          </option>
        </select>
      </div>
      <div
        class="input-group-prepend flex-colunm col-12 col-md-3"
      >
        <label
          class="px-2 fw-bold"
          for="package-type"
        >
          Text
        </label>
        <select
          aria-label="Package Type Filter"
          class="form-select form-select-sm"
          data-testid="package-type-filter"
          id="package-type-filter"
        >
          <option
            value="any"
          >
            Text
          </option>
          <option
            value="jdk"
          >
            JDK
          </option>
          <option
            value="jre"
          >
            JRE
          </option>
        </select>
      </div>
      <div
        class="input-group-prepend flex-colunm col-12 col-md-3"
      >
        <label
          class="px-2 fw-bold"
          for="version"
        >
          Text
        </label>
        <select
          aria-label="Version Filter"
          class="form-select form-select-sm"
          data-testid="version-filter"
          id="version-filter"
        >
          <option
            value="1"
          >
            1 - LTS
          </option>
          <option
            value="2"
          >
            2
          </option>
        </select>
      </div>
    </div>
    <table
      class="table table-bordered releases-table"
      id="download-table"
      style="border-spacing: 0 10px; border-collapse: separate;"
    >
      <thead
        class="table-dark"
      >
        <tr
          class="table-head"
        >
          <td
            class="table-head"
          >
            Text
          </td>
          <td
            class="table-head"
          >
            Text
          </td>
          <td
            class="table-head"
          >
            Text
          </td>
          <td
            class="table-head"
          >
            Text
          </td>
          <td
            class="table-head"
          >
            Text
          </td>
          <td
            class="table-head"
          >
            Text
          </td>
        </tr>
      </thead>
      <tbody
        class="table-light"
      >
        <tr />
      </tbody>
    </table>
    <div
      aria-hidden="true"
      aria-labelledby="checksumModalLabel"
      class="modal fade"
      id="checksumModal"
      style="z-index: 10000;"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-lg"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h5
              class="modal-title"
              id="checksumModalLabel"
            >
              Checksum (SHA256)
            </h5>
            <button
              aria-label="Close"
              class="btn-close"
              data-bs-dismiss="modal"
              type="button"
            />
          </div>
          <div
            class="modal-body text-start"
          >
            <p>
              The checksum is used to ensure the file has not been corrupted during download. You should compare the checksum of the file you received with the value below to ensure the file you received is complete and unmodified.
            </p>
            <p>
              On Windows use 
              <code>
                certUtil -hashfile 
                <var>
                  file
                </var>
                 SHA256
              </code>
              , on Linux use 
              <code>
                sha256sum 
                <var>
                  file
                </var>
              </code>
              , and on macOS use 
              <code>
                shasum -a 256 
                <var>
                  file
                </var>
              </code>
              .
            </p>
            <input
              readonly=""
              style="width: 100%;"
              value=""
            />
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Copy
            </button>
            <button
              class="btn btn-secondary"
              data-bs-dismiss="modal"
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
</main>
`;
